lib60870-C/src/iec60870/cs104/cs104_connection.c

```diff
@@ -860,6 +860,28 @@ handleConnection(void* parameter)
 
     resetConnection(self);
 
+  /*
+   * This guard ensures that the socket doesn't get overwritten by another thread,
+   * which could potentially lead to a double-free error.
```

Comment:
I would add a line about why this could happen, i.e. previous thread isn't successfully destroyed in `CS104_Connection_connectAsync`

Reply:
@dtcaciuc-ggs The previous thread will always get destroyed in `CS104_Connection_connectAsync`, but it's just a matter of when it's destroyed right? And this guard ensures that we don't overwrite the thread before it's destroyed?

Reply:
That function attempts to destroy it but it might not succeed; pthread_join() call inside can fail and that error is not checked. connectAsync would then proceed as if tho thread was destroyed and starts another one using the same "self" structure which then ends up in socket being potentially overwritten.

The hope is that the old thread eventually quits on its own but until then we'll just keep emitting connection failed.

Reply:
Right, I remember you saying that now. Thanks, just added more to the comment.